extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,80 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationLinks();
+		$hasMobileApplicationBadges = count($mobileApplicationsLinks) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationLinks() {
+		$result = [];
+
+		$response = $this->fetchMobileApplicationsDetails();
```

Comment:
It's possible for `fetchMobileApplicationsDetails` to return false, since that's what `Http::request` returns in the case of an error or time out. We should probably check for that.